assets/src/edit-story/panels/size.js

```diff
@@ -34,18 +36,56 @@ function SizePanel( { selectedElements, onSetProperties } ) {
 				label="Width"
 				value={ state.width }
 				isMultiple={ width === '' }
-				onChange={ ( value ) => setState( { ...state, width: isNaN( value ) || value === '' ? '' : parseFloat( value ) } ) }
+				onChange={ ( value ) => {
+					const newWidth = isNaN( value ) || value === '' ? '' : parseFloat( value );
+					setState( {
+						...state,
+						width: newWidth,
+						// @todo: Move to the reducer once available. This is especially critical
+						// because different elements have different aspect ratios.
+						height: typeof newWidth === 'number' && keepRatio && typeof origRatio === 'number' ? newWidth / origRatio : height,
+					} );
+				} }
 				postfix="px"
 				disabled={ isFullbleed }
 			/>
 			<InputGroup
 				label="Height"
 				value={ state.height }
 				isMultiple={ height === '' }
-				onChange={ ( value ) => setState( { ...state, height: isNaN( value ) || value === '' ? '' : parseFloat( value ) } ) }
+				onChange={ ( value ) => {
+					const newHeight = isNaN( value ) || value === '' ? '' : parseFloat( value );
+					setState( {
+						...state,
+						height: newHeight,
+						// @todo: Move to the reducer once available. This is especially critical
+						// because different elements have different aspect ratios.
+						width: typeof newHeight === 'number' && keepRatio && typeof origRatio === 'number' ? newHeight * origRatio : width,
+					} );
+				} }
 				postfix="px"
 				disabled={ isFullbleed }
 			/>
+			<InputGroup
+				type="checkbox"
+				label="Keep ratio"
+				value={ state.keepRatio }
+				isMultiple={ keepRatio === '' }
+				onChange={ ( value ) => {
+					if ( value && typeof origRatio === 'number' ) {
+						onSetProperties( {
+							keepRatio: true,
```

Comment:
Hmm, should the checkbox influence the behavior of Moveable? Perhaps this should only lock the numbers when manually changing?

Reply:
It updates both: the numbers and moveable's properties. I think once we have reducers in place we will move the width/height update there. But we definitely want to update `Moveable` props, such as which resizing corners are allowed, etc.

Reply:
Yep, saw that it's changing both, I'm thinking that maybe it should not change the behavior of Moveable? Maybe the lock icon in the Figma files is for manual updating only? For some reason that's what I'd assume from the UX -- that Moveable works as it does and the lock icon would be for the manual fields only. Anyway, perhaps you know better what's intended for UX.

Reply:
I'll confirm with the @samitron7.

Reply:
@miina I talked to @samitron7 and you were exactly right. These changes shouldn't cause Moveable changes. Further, here are the changes in the latest commit:

1. Ratio "lock" is no longer in the model. It's only part of edit panel.
2. Changing "lock" value does not update the width/height, but simply locks the ratio between them.
3. Moveable is no longer affected by this value either. Per latest discussion with @samitron7, all resize handles are now enabled for the image: corners preserve aspect ratio and the mid-points do not.